usd/src/usd_parser/USD2SDF.cc

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "USD2SDF.hh"
+
+#include "sdf/Console.hh"
+#include "sdf/Types.hh"
+#include "usd_model/WorldInterface.hh"
+#include "USDWorld.hh"
+
+namespace sdf {
+inline namespace SDF_VERSION_NAMESPACE {
+namespace usd {
+/////////////////////////////////////////////////
+USD2SDF::USD2SDF()
+{
+}
+
+/////////////////////////////////////////////////
+USD2SDF::~USD2SDF()
+{
+}
+
+/////////////////////////////////////////////////
+UsdErrors USD2SDF::Read(const std::string &_fileName,
+  tinyxml2::XMLDocument *_sdfXmlOut)
+{
+  UsdErrors errors;
+
+  std::shared_ptr<WorldInterface> worldInterface =
+    std::make_shared<WorldInterface>();
+
+  const auto errorsParseUSD = parseUSDWorld(_fileName, worldInterface);
+  if (!errorsParseUSD.empty())
+  {
+    errors.emplace_back(UsdError(
+      UsdErrorCode::SDF_TO_USD_PARSING_ERROR,
+      "Error parsing usd file [" + _fileName + "]"));
+    return errors;
+  }
+
+  tinyxml2::XMLElement *sdf = nullptr;
```

Comment:
I see that the general approach is to create a tinyxml2 document and manually populate the elements in this function. I suspect this will be harder to scale than first converting to an `sdf::ElementPtr` and calling `ToString`. Even better might be to convert to `sdf::Root`, then call `sdf::Root::ToElement()` then `ToString`. This is much less error prone, handles all the formatting. It's also a lot more future proof. With the current approach, if we update the spec, we have to remember to update this function.